pkg/apis/v1alpha1/staticroute_types.go

```diff
@@ -0,0 +1,77 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type StaticRouteStatusCondition string
+
+const (
+	StaticRouteReady    StaticRouteStatusCondition = "Ready"
+	StaticRouteNotReady StaticRouteStatusCondition = "NotReady"
+)
+
+// StaticRouteCondition defines condition of StaticRoute.
+type StaticRouteCondition struct {
+	// Type defines condition type.
+	Type StaticRouteStatusCondition `json:"type"`
+	// Status defines status of condition type, True or False.
+	Status corev1.ConditionStatus `json:"status"`
+	// Reason shows a brief reason of condition.
+	Reason string `json:"reason,omitempty"`
+	// Message shows a human readable message about condition.
+	Message string `json:"message,omitempty"`
+}
+
+// StaticRouteSpec defines static routes configuration on VPC
+type StaticRouteSpec struct {
+	// Specify next hop routes for network
+	StaticRoutes []StaticRouteItem `json:"routes"`
+}
+
+type StaticRouteItem struct {
+	// Unique route name to indicate the specific node in specific TKG
+	Name string `json:"name"`
+	// Next hop gateway IP address
+	// +kubebuilder:validation:Format=ip
+	Target string `json:"target"`
+	// Specify network address in CIDR format
+	// +kubebuilder:validation:Format=cidr
+	Destination string `json:"destination"`
+}
+
+// StaticRouteStatus defines the observed state of StaticRoute
+type StaticRouteStatus struct {
+	Conditions   []StaticRouteCondition `json:"conditions"`
+	StaticRoutes []StaticRouteItem      `json:"routes"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:storageversion
+
+// StaticRoute is the Schema for the staticroutes API
+type StaticRoute struct {
```

Comment:
Question - what is the corresponding resource type in NSX-T VPC API? I think in most cloud, it is called RouteTable. @lxiaopei

Reply:
NSX called StaticRoute in its API. https://gitreview.eng.vmware.com/c/nsx/+/466368/6/mp/policy/policy-api/src/main/resources/api_spec/root/nsx-policy/policy-vpc-static-routes.yml#31

Reply:
I am not sure I understood the NSX-T API definition correctly. Seems in NSX-T a single route is a "StaticRoute" resource?

But in our CRD definition, "StateRouteSpec" includes a slice of Routes.

Reply:
Yea, the NSX static route API seems a single route to a "network".
So shall we name it RouteTable CRD?

Reply:
the crd definition is not the same as StaticRoute api which a single route includes a slice of next hop.

Reply:
Why we do not use the same definition as NSX-T API? If NSX-T API resource is a single Route (meaning a single destination), we should go the same way. And we should use a slice of next hops too.

Again, please keep all names and structs the same as NSX-T API. We can remove unnecessary fields, but do not add fields, change names or structs.

Reply:
update according to NSXT API

Reply:
done